internal/config/config.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/resource"
+	"sigs.k8s.io/yaml"
+)
+
+// constants for scaffolding version
+const (
+	Version1 = "1"
+	Version2 = "2"
+
+	DefaultPath = "PROJECT"
+)
+
+// Config is the unmarshalled representation of the configuration file
+type Config struct {
+	// Path stores where the config was loaded from
+	path string
+
+	// Version is the project version, defaults to "1" (backwards compatibility)
+	Version string `json:"version,omitempty"`
+
+	// Domain is the domain associated with the project and used for API groups
+	Domain string `json:"domain,omitempty"`
+
+	// Repo is the go package name of the project root
+	Repo string `json:"repo,omitempty"`
+
+	// Resources tracks scaffolded resources in the project
+	// This info is tracked only in project with version 2
+	Resources []GVK `json:"resources,omitempty"`
+
+	// Multigroup tracks if the project has more than one group
+	MultiGroup bool `json:"multigroup,omitempty"`
+}
+
+func exists(path string) (bool, error) {
+	// Look up the file
+	_, err := os.Stat(path)
+
+	// If we could find it the file exists
+	if err == nil {
+		return true, nil
+	}
+
+	// Not existing and different errors are differentiated
+	if os.IsNotExist(err) {
+		err = nil
+	}
+	return false, err
+}
+
+// Exists verifies that the configuration file exists in the default path
+func Exists() (bool, error) {
+	return exists(DefaultPath)
+}
+
+// ExistsIn verifies that the configuration file exists in the provided path
+func ExistsIn(path string) (bool, error) {
+	return exists(path)
+}
+
+func load(path string) (*Config, error) {
```

Comment:
The func will read a file and parse it to return the Config instance?
This file represents the structure. ( because of this the previous name of it was Project)
May :

```suggestion
func newConfigFrom(path string) (*Config, error) {
```

Reply:
See https://github.com/kubernetes-sigs/kubebuilder/pull/1325#discussion_r368285731.